src/plugins/intel_cpu/src/node.h

```diff
@@ -585,6 +587,10 @@ class Node {
         sharedMutex = mutex;
     }
 
+    virtual std::vector<Graph *> getSubGraphs() {
```

Comment:
Discussed that change with @maxnick offline, 
It looks like not a best solution to extend base Node interface with smt that corresponds only to 2 nodes in opset. There are two proposals in fact:
1. Introduce Node derived class that will serve as base class for complex nodes (which includes subgraphs as body) and implemente additional interface on this level, However it might be reasonable only in case some functionality can be reused between IF and TI.
2. Second is to introduce config setter only for TI and IF operaations and use downcast approach as it was done for [sparse rate](https://github.com/usstq/openvino/blob/tq/ti_bf16/src/plugins/intel_cpu/src/graph.cpp#LL348C18-L348C18) for example. In you particular story that seems to be most suitable solution.

Reply:
you mean like this?

```cpp
 void Graph::InitDescriptors() {
       ....
       if (auto tiNodePtr = std::dynamic_pointer_cast<node::TensorIterator>(node)) {
            for (auto &sub_graph : tiNodePtr->getSubGraphs()) {
                sub_graph->setConfig(getConfig());
            }
        }
        if (auto ifNodePtr = std::dynamic_pointer_cast<node::If>(node)) {
            for (auto &sub_graph : ifNodePtr->getSubGraphs()) {
                sub_graph->setConfig(getConfig());
            }
        }
```

I don't like this change because it introduced additional dependency between `graph.cpp` and `TensorIterator` & `If`, I believe that graph should avoid node-type dependent logic. so I come up with another proposal, we can directly set config into node, it's node's choice on how to use the config, it can also solves another issue https://github.com/openvinotoolkit/openvino/pull/12315#issuecomment-1257522279 in which Input node needs to know whether CPU_DENORMALS_OPTIMIZATION  flag is set in config. what do you think?

Reply:
I also found that TI/IF creates subgraph like this:
```
sub_graph.CreateGraph(body, ext_mng, weightCache, sharedMutex);
```
And there are few important parameters missing in the call:
 - config wasn't set (when ExecNetwork create graph, it calls setConfig() before CreateGraph)
 - MultiCache was not reused with parent graph (while it should)
 - DnnlScratchPad was not reused with parent graph (while it should)

This indicates that TI/IF didn't create sub-graph the right way, maybe the CreateGraph() API needs refine to include all important args

Reply:
I followed the `RuntimeEnv` idea, it provides following benefits:
 - instance of `RuntimeEnv` was created in `ExecNetwork::GetGraph` for each stream and it collects all runtime data structures including `Config/extensionManager/WeightCaches/SharedMutex/engine`, and this instance is shared among all nodes and sub-graphs in the same stream. 
 - we can save the `setRuntimeCache/setSharedMutex/setRuntimeScratchPad/...` API calls, this `rtEnv` was passed into Node's constructor and it contains references to all such kind of runtime data structure.
 - we can remove the WA in NodesFactory::create() for calling `setExtManager()` because it now included in `rtEnv`
 - we can add more such common runtime data into `RuntimeEnv` in the future w/o adding new API.
 - the `Graph::CreateGraph()` is changed to require only `RuntimeEnv::Ptr` as additional parameter, `Graph::setConfig()` is removed, now `CreateGraph()` is the only required API to create a graph.
 - subgraphs in TI/IF automatically take benefit of the BF16 config because config is included in `rtEnv`
 - node can access config now, https://github.com/openvinotoolkit/openvino/pull/12315#issuecomment-1257522279 can carry on based on this change.

Reply:
@maxnick Please help to review also, Thx!

Reply:
I think this is good proposal. My initial comment tried to align with existing solution. However as long as you proposed (and already prepared) better solution I believe we can accept it. T
I left some comments. Please take a look. Thanks!